src/StreamableFeesLocker.sol

```diff
@@ -0,0 +1,340 @@
+        uint256 amount1Distributed;
+        address beneficiary;
+        for (uint256 i; i < beneficiaries.length; ++i) {
+            beneficiary = beneficiaries[i].beneficiary;
+            uint256 shares = beneficiaries[i].shares;
+
+            // Calculate share of fees for this beneficiary
+            uint256 amount0 = currency0ToDistribute * shares / WAD;
+            uint256 amount1 = currency1ToDistribute * shares / WAD;
+
+            _distributeFees(beneficiary, poolKey, amount0, amount1);
+
+            amount0Distributed += amount0;
+            amount1Distributed += amount1;
+        }
+
+        // Distribute the remaining fees to the last beneficiary
+        uint256 amount0Remaining =
+            currency0ToDistribute > amount0Distributed ? currency0ToDistribute - amount0Distributed : 0;
+        uint256 amount1Remaining =
+            currency1ToDistribute > amount1Distributed ? currency1ToDistribute - amount1Distributed : 0;
+        _distributeFees(beneficiary, poolKey, amount0Remaining, amount1Remaining);
+
+        // Note: For no-op governance, if recipient is DEAD_ADDRESS (0xdead), the position will be permanently locked
+        // and beneficiaries can collect fees in perpetuity
+        if (block.timestamp >= position.startDate + position.lockDuration && position.recipient != DEAD_ADDRESS) {
+            position.isUnlocked = true;
+
+            // Transfer the position to the recipient
+            ERC721(address(positionManager)).safeTransferFrom(address(this), position.recipient, tokenId, new bytes(0));
+
+            emit Unlock(tokenId, position.recipient);
+        }
+
+        // Update the position in storage
+        positions[tokenId] = position;
+
+        emit DistributeFees(tokenId, currency0ToDistribute, currency1ToDistribute);
+    }
+
+    /// @notice Releases accrued fees to the caller
+    /// @param tokenId ID of the position to release fees from
+    function releaseFees(
+        uint256 tokenId
+    ) external nonReentrant {
+        // Check if position exists
+        PositionData memory position = positions[tokenId];
+        require(position.startDate != 0, PositionNotFound());
+
+        // Check if sender is a beneficiary
+        bool isBeneficiary = false;
+        for (uint256 i = 0; i < position.beneficiaries.length; i++) {
+            if (position.beneficiaries[i].beneficiary == msg.sender) {
+                isBeneficiary = true;
+                break;
+            }
+        }
+        require(isBeneficiary, NotBeneficiary());
+
+        _releaseFees(tokenId, msg.sender);
+    }
+
+    /// @notice Updates currency balances and calculates distributable amounts
+    /// @param position Current position data
+    /// @param poolKey Pool information
+    /// @return amount0ToDistribute Amount of token0 to distribute
+    /// @return amount1ToDistribute Amount of token1 to distribute
+    function _updateCurrencyBalances(
+        PositionData memory position,
```

Comment:
Looks like this param is unused

Reply:
good point thanks!

Reply:
https://github.com/whetstoneresearch/doppler/pull/381/commits/9d44ec255e077df007236e2fda30a626aa7c82e9